### ISSUE TITLE
Find pointmatches using a KDTree

### ISSUE THREAD

**Issue by acardona** [axtimwalde/mpicbg#71]
There are a number of use cases where the approximate position of image tiles is known, and most often, their overlap is very large.

For these cases, comparing SIFT features all to all is a waste of time, and a source of potentially incorrect pointmatches.

Ideally, there would be an extended method with an extra boolean flag to make search be constrained to within a radius. This would turn an n^2 into an n*m, where m << n, reducing computation time enormously.

The one issue I have is: the KDTree implementation is in imglib2, and I understand it's undesirable for mpicbg to depend on imglib2. Short of rolling out a custom solution inside imglib2-algorithm, any suggestions?

**Comment by acardona** [comment]
Thanks very much for reaching out. My easy use case is via `FloatArray2DSIFT.createMatches`, so my question is, what method should be used now to replace that one? It seems that a convenient method like that didn't make it to `FloatArray2DSIFT`.

**Comment by minnerbe** [comment]
You're right in that there is no method using the local matching in `FloatArray2DSIFT`. I'd suggest using `FeatureTransform.matchFeaturesLocally` directly to get access to the local matching.

The only difference is that `FloatArray2DSIFT.createMatches` uses `(f1.scale + f2.scale) / 2.0f)` to weight matches, whereas `FeatureTransform.matchFeaturesLocally` uses a uniform weight of 1. This behavior, however is undocumented, so I'm not sure how much your use case relies on it.

**Comment by acardona** [comment]
Thanks, I'm already doing that. And very good to know there is undocumented behaviour – fortunately I don't weigh the features, so their weights are 1 anyway.

**Comment by acardona** [comment]
By the way `FeatureTransform.matchFeaturesLocally` is brilliant: went from minutes to under a second by being able to specify a radius. Even when big (about 25% of the image width), since it's quadratic, the savings are massive, particularly when extracting over a 1000 features per image.
